### PR TITLE
Use a different thread for each TreehouseApp

### PR DESCRIPTION
**Pull request by swankjesse**
This should prevent different apps from competing with each other for CPU resources, especially during the first few seconds of the host app's launch, when there's a bunch of CPU-bound work to do.

---

- [x] `CHANGELOG.md`'s "Unreleased" section has been updated, if applicable.

### CODE REVIEW

File: redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/RealTreehouseApp.kt
Comment (swankjesse) [inline redwood-treehouse-host/src/commonMain/kotlin/app/cash/redwood/treehouse/RealTreehouseApp.kt:127]: * [ ] Do not merge until this fix lands!

File: redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt
Comment (JakeWharton) [inline redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt:57]: Does this need a dedicated thread? Shouldn't `Disaptchers.IO` be sufficient? Same for iOS.

**Review by JakeWharton — APPROVED** [review]

File: redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt
Comment (swankjesse) [reply redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt:57]: Unfortunately Dispatchers.IO isn’t thread-confined, and I need that for SQLite.

Comment (swankjesse) [reply redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt:57]: I might be able to borrow a thread from that pool though. Somehow.

Comment (JakeWharton) [reply redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt:57]: You can limit the parallelism to 1 if concurrency/parallelism is the problem, but that doesn't guarantee the same thread is used.

Comment (swankjesse) [reply redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt:57]: Yeah, that could work. Lemme try.

Comment (swankjesse) [reply redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt:57]: (In follow-up, as I do.)

**Comment by dnagler** [comment]
Belated LGTM, thanks for taking this on!

File: redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt
Comment (swankjesse) [reply redwood-treehouse-host/src/androidMain/kotlin/app/cash/redwood/treehouse/AndroidTreehouseDispatchers.kt:57]: https://github.com/cashapp/redwood/pull/2188/files

**Comment by hunterestrada** [comment]
Yeah, this is great!

**Pull request merged**
